gum/backend-arm64/gumstalker-arm64.c

```diff
@@ -439,6 +441,45 @@ gum_stalker_finalize (GObject * object)
   G_OBJECT_CLASS (gum_stalker_parent_class)->finalize (object);
 }
 
+static GumVirtualizationRequirements
+gum_exec_block_virtualize_linux_sysenter_insn (GumExecBlock * block,
+                                               GumGeneratorContext * gc)
+{
+  GumArm64Writer *cw = gc->code_writer;
+  gconstpointer continue_normally = cw->code + 1;
+  gconstpointer end_payload = cw->code + 3;
+  const cs_insn * insn = gc->instruction->ci;
+  const guint32 mrs_x15_nzcv = 0xd53b420f;
+  const guint32 msr_nzcv_x15 = 0xd51b420f;
+
+  gum_arm64_relocator_skip_one (gc->relocator);
+
+  if (gc->opened_prolog != GUM_PROLOG_NONE)
+    gum_exec_block_close_prolog (block, gc);
+
+  gum_arm64_writer_put_instruction (cw, mrs_x15_nzcv);
+
+  gum_arm64_writer_put_sub_reg_reg_imm (cw, ARM64_REG_X17,
+      ARM64_REG_X8, __NR_clone);
+  gum_arm64_writer_put_cbnz_reg_label (cw, ARM64_REG_X17, continue_normally);
```

Comment:
Are we allowed to clobber `X17` here?

Reply:
X17 is a scratch register and it's normally used for such purposes for the linker - in fact when doing gum_exec_block_write_exec_generated_code it uses the same, X16 and X17 to return to the next instruction. X15 was clobbered and fixed, but when transfering to the original code X17 has to be clobbered.

Reply:
If there is another way to return back to the original code I am all ears

Reply:
I know it's a scratch register, but if we clobber it someone might write some inline assembly to detect Stalker this way.

We don't actually clobber it in `write_exec_generated_code()`, we generate code that saves it before we clobber it.